Frameworks/UIKit/UITextField.mm

```diff
@@ -100,6 +100,10 @@ - (void)setText:(NSString*)text {
             _passwordBox.password = _text;
         } else {
             _textBox.text = _text;
+            // Ensure caret at end of field in case we programmatically
+            // gain focus (becomeFirstResponder) after the text is set:
+            _textBox.selectionStart = [_text length];
+            _textBox.selectionLength = 0;
```

Comment:
PasswordBox doesn't have the SelectionStart/Length properties. However it already implements the right functionality automatically. (I did the basic test which is to programmatically set the text then programmatically gain focus with a secureMode UITextField - the caret just appears at the end)